libs/labelbox/src/labelbox/schema/task.py

```diff
@@ -214,3 +227,169 @@ def get_task(client, task_id):
+            query=query_str,
+            params=params,
+            dereferencing=['successesfulDataRowImports', 'nodes'],
+            obj_class=lambda _, data_row: {
+                'id': data_row.get('id'),
+                'external_id': data_row.get('externalId'),
+                'row_data': data_row.get('rowData'),
+                'global_key': data_row.get('globalKey'),
+            },
+            cursor_path=['successesfulDataRowImports', 'after'],
+        )
+
+    def _download_errors_paginated(self) -> PaginatedCollection:
+        page_size = DOWNLOAD_RESULT_PAGE_SIZE  # hardcode to avoid overloading the server
+        from_cursor = None
+
+        query_str = """query FailedDataRowImportsPyApi($taskId: ID!, $first: Int, $from: String)  {
+                    failedDataRowImports(data: { taskId: $taskId, first: $first, from: $from})
+                        {
+                            after
+                            total
+                            results {
+                                message
+                                spec {
+                                    externalId
+                                    globalKey
+                                    rowData
+                                        metadata {
+                                            schemaId
+                                            value
+                                            name
+                                        }
+                                        attachments {
+                                            type
+                                            value
+                                            name
+                                        }                                    
+                                }
+                            }
+                        }
+                    }
+                """
+
+        params = {
+            'taskId': self.uid,
+            'first': page_size,
+            'from': from_cursor,
+        }
+
+        def convert_errors_to_legacy_format(client, data_row):
+            spec = data_row.get('spec', {})
+            return {
+                'message':
+                    data_row.get('message'),
+                'failedDataRows': [{
+                    'externalId': spec.get('externalId'),
+                    'rowData': spec.get('rowData'),
+                    'globalKey': spec.get('globalKey'),
+                    'metadata': spec.get('metadata', []),
+                    'attachments': spec.get('attachments', []),
+                }]
+            }
+
+        return PaginatedCollection(
+            client=self.client,
+            query=query_str,
+            params=params,
+            dereferencing=['failedDataRowImports', 'results'],
+            obj_class=convert_errors_to_legacy_format,
+            cursor_path=['failedDataRowImports', 'after'],
+        )
+
+    def _results_as_list(self) -> Optional[List[Dict[str, Any]]]:
```

Comment:
nit: I believe `PaginatedCollection` is an iterator, in which case you can reduce this function to a single line of code (or remove it entirely).

```python
def _results_as_list(self) -> Optional[List[Dict[str, Any]]]:
    return list(self._download_results_paginated()) or None
```